internal/component/redis/settings.go

```diff
@@ -0,0 +1,103 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation and Dapr Contributors.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package redis
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/dapr/components-contrib/internal/config"
+)
+
+type Settings struct {
+	// The Redis host
+	Host string `mapstructure:"redisHost"`
```

Comment:
Is this structure backwards compatible in the sense that existing components from apps will continue to work as-is?

Reply:
Yes, benefiting from the configuration of these components is relatively small, and most of the naming rules are consistent. There are also two configs have different metadata name for the same configuration, but I've handled this by parsing the old metadata first.